cli/config.ts

```diff
@@ -0,0 +1,46 @@
+import { exists, mkdir, readFile, writeFile } from "fs";
+import { homedir } from "os";
+import { join } from "path";
+import { promisify } from "util";
+
+export interface IConfigSettings {
+  allowAnonymousAnalytics?: boolean;
+  userId?: string;
+}
+
+export function getConfigDirPath() {
+  return join(homedir(), ".dataform");
+}
+
+export function getConfigSettingsPath() {
+  return join(getConfigDirPath(), "settings.json");
+}
+
+export async function getConfigSettings(): Promise<IConfigSettings> {
+  try {
+    if (await promisify(exists)(getConfigSettingsPath())) {
```

Comment:
use `fs-extra`, it's a bajillion times better

Reply:
I'm really trying to limit the number of packages we use here given that these are packages we actually publish.

Reply:
- fs-extra has a total of 3 dependencies
- the user installs the dataform CLI once and then updates rarely, it's not a big deal (like it might be for @dataform/core)